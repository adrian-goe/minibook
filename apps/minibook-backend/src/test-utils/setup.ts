import { INestApplication } from '@nestjs/common';
import request from 'supertest';
import { Test, TestingModule } from '@nestjs/testing';
import { AppModule } from '../app/app.module';

export async function gqlTestQuery(
  app: INestApplication,
  query: string,
  variables: unknown = {}
) {
  return request(app.getHttpServer()).post('/graphql').send({
    query,
    variables,
  });
}

export async function setupTestApp() {
  const moduleFixture: TestingModule = await Test.createTestingModule({
    imports: [AppModule],
  }).compile();
  return moduleFixture.createNestApplication();
}
