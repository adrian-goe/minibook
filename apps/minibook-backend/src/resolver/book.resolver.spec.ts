import { INestApplication } from '@nestjs/common';
import { gqlTestQuery, setupTestApp } from '../test-utils/setup';

describe('Book Resolver', function () {
  let app: INestApplication;

  beforeAll(async () => {
    app = await setupTestApp();
    await app.init();
  });

  afterAll(async () => {
    await app.close();
  });

  it('should return with books', async () => {
    const response = await gqlTestQuery(
      app,
      `
query BookQuery {
  getBooks {
    id
    isbn
    name
    author {
      name
    }
  }
}
`
    );

    expect(response.status).toEqual(200);
    expect(response.body.data).toMatchSnapshot();
  });

  xit('should create a book', async () => {
    const response = await gqlTestQuery(
      app,
      `
mutation Mutation($createBook: BookCreateInput!) {
  createBook(createBook: $createBook) {
    author {
      name
    }
    isbn
    name
    id
  }
}
`,

      {
        createBook: {
          name: 'Momo',
          isbn: '978-3522202107',
        },
      }
    );

    expect(response.status).toEqual(200);
    expect(response.body.data).toMatchSnapshot();
  });
});
