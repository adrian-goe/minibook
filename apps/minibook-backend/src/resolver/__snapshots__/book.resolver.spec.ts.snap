// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Resolver should return with books 1`] = `
{
  "getBooks": [
    {
      "author": {
        "name": "John Smith",
      },
      "id": "5291a7fc-d663-40d6-935a-38b52d697fe3",
      "isbn": 123456789,
      "name": "Book Title 1",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "e2961f44-9b67-4eb5-8924-0bc047647317",
      "isbn": 987654321,
      "name": "Book Title 2",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "f3e92efe-d784-4dce-bd4e-09e105239e75",
      "isbn": 159753486,
      "name": "Book Title 3",
    },
    {
      "author": {
        "name": "John Smith",
      },
      "id": "ca3a92b0-d386-4c26-8715-726a884f6b59",
      "isbn": 159753486,
      "name": "Book Title 4",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "0067b91f-d513-4877-b1e5-c7d9217ba993",
      "isbn": 159753486,
      "name": "Book Title 5",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "d0c347bc-1545-4f38-bf82-e71a05b429c4",
      "isbn": 159753486,
      "name": "Book Title 6",
    },
    {
      "author": {
        "name": "John Smith",
      },
      "id": "57443459-d95d-4911-b2df-7bcfc3f0c7f0",
      "isbn": 159753486,
      "name": "Book Title 7",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "5e0ddd47-abdf-4c75-9174-2dfdbbbfb29d",
      "isbn": 159753486,
      "name": "Book Title 8",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "340f0fcb-ca7f-4037-98b3-7f4bfea2eb08",
      "isbn": 159753486,
      "name": "Book Title 9",
    },
    {
      "author": {
        "name": "John Smith",
      },
      "id": "ce9a57b4-3e9d-41b3-b306-3d2140b2874a",
      "isbn": 159753486,
      "name": "Book Title 10",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "3e58038e-fcd3-4125-b036-392762547d4e",
      "isbn": 159753486,
      "name": "Book Title 11",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "dc24b466-9083-4159-b094-5d3eb25bfad7",
      "isbn": 159753486,
      "name": "Book Title 12",
    },
    {
      "author": {
        "name": "John Smith",
      },
      "id": "2dbbe6d9-95cb-427d-b28a-ab5f4f6c8f50",
      "isbn": 159753486,
      "name": "Book Title 13",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "b7e46666-2385-41a3-8899-7740cc3ab7b1",
      "isbn": 159753486,
      "name": "Book Title 14",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "32e156fb-9e26-4b03-83a3-9697ad1f3897",
      "isbn": 159753486,
      "name": "Book Title 15",
    },
    {
      "author": {
        "name": "John Smith",
      },
      "id": "dd00daa7-1afc-416f-a4f3-5dbf22852e43",
      "isbn": 159753486,
      "name": "Book Title 16",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "85d8380d-e7ae-483a-9850-d51ad1882d9e",
      "isbn": 159753486,
      "name": "Book Title 17",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "7f040fa8-f690-4138-ac46-ab1e244f87bd",
      "isbn": 159753486,
      "name": "Book Title 18",
    },
    {
      "author": {
        "name": "John Smith",
      },
      "id": "9ec83d45-532f-4f10-8b87-eac8192805e8",
      "isbn": 159753486,
      "name": "Book Title 19",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "f5116dff-528c-41d2-917c-96ca2f32bf2f",
      "isbn": 159753486,
      "name": "Book Title 20",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "16fbcf5b-05c5-4c0c-ac1b-8563e8623313",
      "isbn": 159753486,
      "name": "Book Title 21",
    },
    {
      "author": {
        "name": "John Smith",
      },
      "id": "14b7db0a-ec3b-460e-ab97-f94eb2ec6233",
      "isbn": 159753486,
      "name": "Book Title 22",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "a28c4b71-6f86-40a7-85ac-3d4c1454c07b",
      "isbn": 159753486,
      "name": "Book Title 23",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "0c163fef-88dd-4685-b1c8-9fcd589c78ed",
      "isbn": 159753486,
      "name": "Book Title 24",
    },
    {
      "author": {
        "name": "John Smith",
      },
      "id": "ac908521-af52-4eba-9fc0-b8746b5f7522",
      "isbn": 159753486,
      "name": "Book Title 25",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "33c4d945-9758-4275-a938-5bb2027f58c6",
      "isbn": 159753486,
      "name": "Book Title 26",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "befb14c9-c91a-47d4-8a6e-220ee6debc5f",
      "isbn": 159753486,
      "name": "Book Title 27",
    },
    {
      "author": {
        "name": "John Smith",
      },
      "id": "e3843410-2a7b-4591-83de-3fbe6dd1e641",
      "isbn": 159753486,
      "name": "Book Title 28",
    },
    {
      "author": {
        "name": "Jane Doe",
      },
      "id": "55ada36a-65d0-4d0c-9c26-e79b7368d4b0",
      "isbn": 159753486,
      "name": "Book Title 29",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "30d3094f-5092-41c9-b64c-cbfdf92df7d1",
      "isbn": 159753486,
      "name": "Book Title 30",
    },
    {
      "author": {
        "name": "William Brown",
      },
      "id": "513ee25e-608e-41dc-8274-12b6276b383b",
      "isbn": 159753486,
      "name": "Book Title 31",
    },
  ],
}
`;
